src/main/webapp/jsp/notification.jsp

```diff
@@ -0,0 +1,34 @@
+<%@ page import="java.io.PrintWriter" %>
+<%@ page import="java.util.Map" %>
+<%@ page import="com.todolist.app.model.Item" %>
+<%@ page import="java.util.Optional" %>
+<%@ page import="com.todolist.app.service.ToDoStoreSingleton" %>
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Notification</title>
+</head>
+<body>
+<h1>Information about task</h1></h1>
```

Comment:
I looks better now than it was. But it's difficult to reuse. You should avoid having too much java code within you jsp page. I mean it should contain some code, but just to represent the notification. E.g. to this jsp page you might pass the final list of task in addition to operation and and id.

Can we assign all necessary info to the request and fetch from it here. I mean transform our page to look like this https://www.geeksforgeeks.org/getattribute-passing-data-from-server-to-jsp/

```
<%@page import="saagnik.Student"%>
<%@page import="java.util.ArrayList"%>
<%@page contentType="text/html" pageEncoding="UTF-8"%>
<!DOCTYPE html>
<html>
  <head>
   <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
   <title>Student List</title>
  </head>
  <body>
      <h1>Displaying Student List</h1>
      <table border ="1" width="500" align="center">
         <tr bgcolor="00FF7F">
          <th><b>Student Name</b></th>
          <th><b>Student Age</b></th>
          <th><b>Course Undertaken</b></th>
         </tr>
        <%-- Fetching the attributes of the request object
             which was previously set by the servlet 
              "StudentServlet.java"
        --%> 
        <%ArrayList<Student> std = 
            (ArrayList<Student>)request.getAttribute("data");
        for(Student s:std){%>
        <%-- Arranging data in tabular form
        --%>
            <tr>
                <td><%=s.getName()%></td>
                <td><%=s.getAge()%></td>
                <td><%=s.getCrs()%></td>
            </tr>
            <%}%>
        </table> 
        <hr/>
    </body>
</html>
```


You see code above could be reused not only for ADD operation but also for UPDATE since it has common structure and lose coupled.